hyperspy/axes.py

```diff
@@ -655,6 +720,18 @@ def _append_axis(self, *args, **kwargs):
         axis.axes_manager = self
         self._axes.append(axis)
 
+    def _on_index_changed(self):
+        self._update_attributes()
+        self.events.indices_changed.trigger(axes_manager=self)
+
+    def _on_view_changed(self):
+        self._update_attributes()
+        self.events.axes_changed.trigger()
```

Comment:
There are a ton of potential useful Events in hyperspy, that's why this PR is really useful. However, I don't think its this PRs job to introduce all of them, as that will require a lot of forethought and analysis. This is the only event I'm introducing that does not replace an existing behavior in hyperspy. It might be more appropriate to remove it here, and then add it in the `interactive` PR though?